src/org/thoughtcrime/securesms/ShareActivity.java

```diff
@@ -200,6 +200,19 @@ public void onMediaResolved(Uri uri) {
   }
 
   private void handleResolvedMedia(Intent intent) {
+
+    if(getIntent().getStringArrayExtra(Intent.EXTRA_EMAIL) != null)
+    {
+      final String addr = getIntent().getStringArrayExtra(Intent.EXTRA_EMAIL)[0];
+      int contactId = dcContext.lookupContactIdByAddr(addr);
+
+      if(contactId == 0)
+        contactId = dcContext.createContact(null, addr);
+
+      int chatId = dcContext.createChatByContactId(contactId);
+      intent.putExtra(EXTRA_CHAT_ID, chatId);
+    }
+
     int       chatId           = intent.getIntExtra(EXTRA_CHAT_ID, -1);
```

Comment:
Concerning usability, this definitely improves things, good job!
My proposal for improvement:
```suggestion
    int       chatId           = intent.getIntExtra(EXTRA_CHAT_ID, -1);

    String[] extraEmail = getIntent().getStringArrayExtra(Intent.EXTRA_EMAIL);
    if(chatId == -1 && extraEmail != null && extraEmail.length > 0) {
      final String addr = extraEmail[0];
      int contactId = dcContext.lookupContactIdByAddr(addr);

      if(contactId == 0)
        contactId = dcContext.createContact(null, addr);

      chatId = dcContext.createChatByContactId(contactId);
    }
```
The main thing was that this code crashed when `getIntent().getStringArrayExtra(Intent.EXTRA_EMAIL)` returned an empty array (ArrayIndexOutOfBoundsException).

Reply:
no objections, bugs must die :)

Reply:
Now there is this unchecked indexing again. (the code will throw an exception if getStringArrayExtra() returns null)

Reply:
sorry, I committed your suggestion before, but looks like something went wrong :(